src/components/ArticlePage/index.js

```diff
@@ -95,22 +95,48 @@ export class ArticlePage extends React.Component {
     });
   }
 
+  handleCommentInput(event) {
+    this.setState({
+      [event.target.name]: event.target.value
+    });
+  }
+
+  handleCommentPost() {
+    const { commentInput } = this.state;
+    const { payload, postArticleComment } = this.props;
+    postArticleComment(payload.id, commentInput);
+    this.setState({
+      commentInput: ''
+    });
+  }
+
   render() {
     const {
-      articleLikeStatus, bookmarkState, commentLikeState, reportTypes
+      articleLikeStatus, bookmarkState, commentLikeState, reportTypes, commentInput,
     } = this.state;
-    let { article } = this.state;
-    const { payload } = this.props;
-    article = payload;
-    const articleBody = HtmlToReactParser.parse(article.body);
+
+    const { payload, isLoggedIn } = this.props;
+
+    const article = payload;
+
+    // get author information
     const { author } = article;
+
+    // parse html article body to react
+    const articleBody = HtmlToReactParser.parse(article.body);
+
+    // get user information from local storage
+    const user = JSON.parse(localStorage.getItem('user'));
+
```

Comment:
What happens if a user does not exist on the local storage? This will crash the app. Do well to cover all edge cases. Also, this is a repetition of the getToken() helper function, you might consider reusing that.

Reply:
Have you done your due diligence to test that this actually crashes the app, because it doesn't. If a user does not exist in local storage, that will resolve to undefined.

It is also not a repetition of the `getToken` helper function  which get the user's `token`. I am getting the user object here, not just the token.

Reply:
> Have you done your due diligence to test that this actually crashes the app, because it doesn't. If a user does not exist in local storage, that will resolve to undefined.
> 
Yes since line 129 fails silently to `undefined` or even `null`
What happens to the app when the program flow gets to line 219.
Is interpreting `undefined.fullName` or `null.fullName` not fatal?

<img width="1361" alt="screen shot 2019-01-01 at 3 01 52 pm" src="https://user-images.githubusercontent.com/30579559/50573596-9cb8d780-0dd7-11e9-989e-41a4967ef29c.png">
Please fix the above hidden bug as contained in the snapshot. Also writing production codes also means we have to write defensively.

Reply:
Also line 129  is an exact replica of line 2 of the getToken file. I am of a strong opinion that that line 2 can be reused across boards. Even at that, we have a utils folder where we access our localStorage facilites. This should not be done inside a component.
<img width="385" alt="screen shot 2019-01-01 at 3 28 44 pm" src="https://user-images.githubusercontent.com/30579559/50573766-5c0e8d80-0dda-11e9-8d29-f766567d3a66.png">

Reply:
fixed